baseplate/events/queue.py

```diff
@@ -39,24 +39,34 @@ def __init__(self, topic, event_type, timestamp=None, id=None):
         self.id = id or uuid.uuid4()
         self.payload = {}
 
-    def get_field(self, key, obfuscated=False):
+    def get_field(self, key):
         """Get the value of a field in the event.
 
         If the field is not present, :py:data:`None` is returned.
 
         :param str key: The name of the field.
-        :param bool obfuscated: Whether to look for the field in the obfuscated
-            payload.
 
         """
 
-        if not obfuscated:
-            payload = self.payload
-        else:
-            payload = self.payload.get("obfuscated_data", {})
-        return payload.get(key, None)
+        payload = self.payload
+        obfuscated_data = self.payload.get("obfuscated_data", {})
```

Comment:
I'm wondering if I made a bad choice in mimicking the serialized form of the event while building it and keeping the obfuscated fields in the regular payload dict. As a crazy alternative, and assuming we have an enum of field types like mentioned down in `set`, what if payload became a dict from type -> fields, so `set` would just do `self.payloads_by_type[field_type][key] = value` (with sanity checking) and get would just loop through the payload types looking for the key in each dict. `serialize` would need a bit more work, but overall I think it might be a bit cleaner?